frontend/src/components/new-widget-wizard.js

```diff
@@ -0,0 +1,321 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  Checkbox,
+  Form,
+  FormGroup,
+  Grid,
+  GridItem,
+  Modal,
+  ModalVariant,
+  Radio,
+  Stack,
+  StackItem,
+  Text,
+  TextArea,
+  TextContent,
+  TextInput,
+  Title,
+  Wizard
+} from '@patternfly/react-core';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+} from '@patternfly/react-table';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+
+import { Settings } from '../settings';
+
+export class NewWidgetWizard extends React.Component {
+  static propTypes = {
+    dashboard: PropTypes.object,
+    onSave: PropTypes.func,
+    onClose: PropTypes.func,
+    isOpen: PropTypes.bool
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      widgetTypes: [],
+      title: '',
+      params: {},
+      weight: 0,
+      isTitleValid: false,
+      selectedType: null,
+      selectedTypeId: null,
+      stepIdReached: 1
+    };
+  }
+
+  onSave = () => {
+    const { params } = this.state;
+    let newParams = {};
+    Object.entries(params).forEach(paramPair => {
+      const param = this.state.selectedType.params.find(el => el.name === paramPair[0]);
+      if (param) {
+        // Some values need to be typecast/parsed
+        let newValue = paramPair[1];
+        if (param.type === 'float') {
+          newValue = parseFloat(newValue);
+        }
+        else if (param.type === 'integer') {
+          newValue = parseInt(newValue);
+        }
+        else if (param.type === 'list') {
+          newValue = newValue.split('\n').map(val => val.trim());
+        }
+        newParams[paramPair[0]] = newValue;
+      }
+    });
+    const newWidget = {
+      title: this.state.title,
+      params: newParams,
+      weight: parseInt(this.state.weight),
+      type: 'widget',
+      widget: this.state.selectedTypeId
+    };
+    if (this.props.dashboard) {
+      newWidget.dashboard_id = this.props.dashboard.id;
+      newWidget.project_id = this.props.dashboard.project_id;
+    }
+    this.props.onSave(newWidget);
+    this.setState({
+      title: '',
+      params: [],
+      weight: 0,
+      isTitleValid: false,
+      selectedType: null,
+      selectedTypeId: null,
+      stepIdReached: 1
+    });
+  }
+
+  onClose = () => {
+    this.setState({
+      title: '',
+      params: [],
+      weight: 0,
+      isTitleValid: false,
+      selectedType: null,
+      selectedTypeId: null,
+      stepIdReached: 1
+    });
+    this.props.onClose();
+  }
+
+  onSelectType = (_, event) => {
+    let selectedTypeId = event.currentTarget.value;
+    let selectedType = null;
+    let params = {};
+    this.state.widgetTypes.forEach(widgetType => {
+      if (widgetType.id === selectedTypeId) {
+        selectedType = widgetType;
+        widgetType.params.forEach(param => {
+          let paramDefault = '';
+          if (param.type === 'integer' || param.type === 'float') {
+            paramDefault = 0;
```

Comment:
For `weight` it makes sense to have this be `0` but for the other parameters, `0` will return no data. I did this by accident just testing it and I think other people might too.

Reply:
I think that in future sometime we should add default values to parameters.

Reply:
that sounds good